packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -634,14 +639,32 @@ class DevFS {
         if (!content.isModified || bundleFirstUpload) {
           return;
         }
+        // Modified shaders must be recompiled per-target platform.
         final Uri deviceUri = _fileSystem.path.toUri(_fileSystem.path.join(assetDirectory, archivePath));
         if (deviceUri.path.startsWith(assetBuildDirPrefix)) {
           archivePath = deviceUri.path.substring(assetBuildDirPrefix.length);
         }
-        dirtyEntries[deviceUri] = content;
-        syncedBytes += content.size;
-        if (archivePath != null && !bundleFirstUpload) {
-          assetPathsToEvict.add(archivePath);
+
+        if (bundle.entryKinds[archivePath] == AssetKind.shader) {
+          final Future<DevFSContent?> pending = shaderCompiler.recompileShader(content);
```

Comment:
This is just going to recompile the dirty ones, right? Otherwise, this should use a pool.

Reply:
Added a pool to the development compiler